### PR TITLE
Move Antiforgery.GetHtml() to MVC as an extension

### PR DESCRIPTION
**Pull request by rynowak**
/cc @dougbu

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/AntiforgeryExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                throw new ArgumentNullException(nameof(antiforgery));
+            }
+
+            if (httpContext == null)
+            {
+                throw new ArgumentNullException(nameof(httpContext));
+            }
+
+            var tokenSet = antiforgery.GetAndStoreTokens(httpContext);
+            return new InputContent(tokenSet);
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/AntiforgeryExtensions.cs]: Pretty much moved as-is, the only changes were to the constructor of `InputContent`

File: test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ViewFeatures/AntiforgeryExtensionsTest.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+using Microsoft.AspNetCore.Mvc.TestCommon;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures
+{
+    /// <summary>
+    /// Extension methods for <see cref="IAntiforgery"/>.
+    /// </summary>
+    public class AntiforgeryExtensionsTest
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ViewFeatures/AntiforgeryExtensionsTest.cs]: I didn't copy the existing tests, they test a bunch of invariants of the antiforgery system that are maintained by us calling `GetAndStoreTokens`

Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ViewFeatures/AntiforgeryExtensionsTest.cs]: That's fine. MVC is also already testing this code path a fair amount in functional tests.

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/AntiforgeryExtensions.cs (hunk 1 above)
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/AntiforgeryExtensions.cs]: WFM

**Comment by dougbu** [comment]
:shipit:

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/AntiforgeryExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                        .Append(_requestToken)
+                        .AppendHtml("\" />");
+                }
+
+                writer.Write("<input name=\"");
+                encoder.Encode(writer, _fieldName);
+                writer.Write("\" type=\"hidden\" value=\"");
+                encoder.Encode(writer, _requestToken);
+                writer.Write("\" />");
+            }
```
Comment (javiercn) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/AntiforgeryExtensions.cs]: Most likely I'm missing something, but won't this method write the tag twice in the case writer is an IHtmlContentBuilder?

**Comment by javiercn** [comment]
Looks good to me :shipit: but make sure you build before you check-in

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/AntiforgeryExtensions.cs (hunk 3 above)
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/AntiforgeryExtensions.cs]: Yeah probably. #YOLO

**Pull request closed without merging**